File: 351004/Brazhalovich/Lab_1/src/main/java/org/example/newsapi/dto/response/NewsResponseTo.java
package org.example.newsapi.dto.response;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Data
public class NewsResponseTo {
    private Long id;


    //@JsonProperty("user")
    //@JsonAlias({"userId", "user"})
    private Long userId;

    private String title;
    private String content;
    private LocalDateTime created;
    private LocalDateTime modified;

    //@JsonProperty("marker")
    private Set<Long> markerIds = new HashSet<>();

    public Long getUser() {
        return this.userId;
    }

    // Jackson создаст поле "marker" в JSON ответе
    public Set<Long> getMarker() {
        return this.markerIds;
    }
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/dto/news/NewsResponseTo.kt
package com.distcomp.dto.news

import java.io.Serializable
import java.time.LocalDateTime

data class NewsResponseTo(
    val id: Long,
    val title: String,
    val content: String,
    val created: LocalDateTime,
    val modified: LocalDateTime,
    val userId: Long
) : Serializable
